### PR TITLE
#12253: Support for Batch normalization

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/12253
https://github.com/tenstorrent/tt-metal/issues/15372

### Problem description
Provide support for batch normalization 
https://pytorch.org/docs/stable/generated/torch.nn.functional.batch_norm.html

### What's changed

- Provided support for batch normalization 
- `data_gen_with_range_batch_norm ` to generate shape as in pytorch and pad the inputs to pass to `ttnn::batch_norm`
- In Progress task : 
  - Return running stats when in training
  - Adding sweep test
  - Move from eltwise to normalization folder

### Checklist
- [x] [Post commit CI](https://github.com/tenstorrent/tt-metal/actions/runs/12121027313) passes

### CODE REVIEW

**Review by Aswinmcw — APPROVED** [review]
LGTM

**Review by mouliraj-mcw — APPROVED** [review]

File: tests/ttnn/unit_tests/operations/eltwise/test_batch_norm.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        running_mean=mean_tensor,
+        running_var=var_tensor,
+        weight=weight_tensor,
+        bias=bias_tensor,
+        training=True,
+    )
+
+    # comments will be removed befor merge
+
+    # ttnn.set_printoptions(profile="full")
```
Comment (umadevimcw) [inline tests/ttnn/unit_tests/operations/eltwise/test_batch_norm.py:59]: Add sweep test as well

**Review by umadevimcw — CHANGES_REQUESTED** [review]

**Comment by VirdhatchaniKN** [comment]
Closing PR as device version is merged to main
